src/aus.h

```diff
@@ -0,0 +1,329 @@
+        {
+            if (_mp.find( key ) != _mp.end() && _mp[key] >0)
+            {
+                _mp[key]->decrement();
+            }
+        }
+
+        long get(std::string key)
+        {
+            assert_key_exists(key);
+            return _mp[key]->get();
+        }
+
+        void start(std::string key)
+        {
+            if (_mp.find(key) == _mp.end())
+            {
+                _mp.insert(std::make_pair(key, new aus_timer()));
+                _mp[key]->start();
+            }
+            else
+            {
+                _mp[key]->start();
+            }
+        }
+
+        void stop(std::string key)
+        {
+            assert_key_exists(key);
+            _mp[key]->stop();
+        }
+
+        std::vector<std::string> get_counters_list()
+        {
+            std::vector<std::string> result;
+            for (const auto& pair : _mp) {
+                result.push_back(pair.first);
+            }
+            return result;
+        }
+
+        bool device_exist( std::string serial )
+        {
+            if ( _mp_devices_manager.find( serial ) == _mp_devices_manager.end())
+            {
+                return false;
+            }
+            return true;
+        }
+
+        void insert_device_to_device_manager( std::string serial, std::string name )
+        {
+            increment( librealsense::aus_build_system_counter_name( "CONNECTED_DEVICES" ));
+            increment( librealsense::aus_build_system_counter_name( "CONNECTED_DEVICES", name ) );
+            _mp_devices_manager[serial]=name;
+        }
+
+        void on_device_changed(std::string serial, std::string name )
+        {
+            if ( !device_exist( serial ) )
+            {
+                insert_device_to_device_manager( serial, name );
+            }
+        } 
+
+    private:
+        std::unordered_map<std::string, aus_value*> _mp;
+        std::unordered_map<std::string, std::string > _mp_devices_manager;
+
+        long _start_time;
+
+        std::string _librealsense_version;
+        std::string _os_name;
+        std::string _platform_name;
+
+        const char * PLATFORM =
+
+            #ifdef _WIN64
+            "Windows amd64";
+        #elif _WIN32
+            "Windows x86";
+        #elif __linux__
+            #ifdef __arm__
+            "Linux arm";
+        #else
+            "Linux amd64";
+        #endif
+        #elif __APPLE__
+            "Mac OS";
+        #elif __ANDROID__
+            "Linux arm";
+        #else
+            "";
+        #endif
+
+        void assert_key_exists(std::string key)
+        {
+            if (_mp.find(key) == _mp.end())
+            {
+                throw std::runtime_error("\"" + key + "\" does not exist");
+            }
+        }
+
+    }; // end of class aus_data
+
+#else
+    class aus_data
+    {
+    public:
+
+        void set(std::string key, long value)
+        {
+            throw std::runtime_error("set is not supported without BUILD_AUS");
```

Comment:
You can create a MACRO that get the function signature and throw the proper exception .
Then you will have a short single line per function list of unsupported functions.
Not sure I used the # properly but something like this:
```
#define NOT_SUPPORTED(func_api) /
##func_api## {throw std::runtime_error("function ##func_api## is not supported without BUILD_AUS flag on");}
```
`NOT_SUPPORTED("void set(std::string key, long value)")`

Reply:
done